wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/Sections/ConversationEncryptionProtocolSectionController.swift

```diff
@@ -43,6 +43,7 @@ extension ConversationEncryptionProtocolSectionController {
     }
 
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        tapAction?()
+        let cell = collectionView.cellForItem(at: indexPath)!
+        tapAction(cell)
```

Comment:
Why do force unwrap here? Can we do something like:

```swift
if let cell = collectionView.cellForItem(at: indexPath) {
    tapAction(cell)
} else {
    // Handle the case where the cell is nil, if necessary
}
```

Reply:
Sometimes I consider crashing hard better than failing silently and this is one example:

On tapping a cell the collection view passes the indexPath of that cell as argument to this delegate method. So at this point we know that the cell exists and that force-unwrapping is safe.

If an invalid indexPath is passed it will be an API violation and I consider failing hard the proper consequence so that the issue is detected early, before the app is shipped.

Reply:
I personally agree with @caldrian here. I think in some cases it is fine to force unwrap or fatal error so we actually know something is going wrong rather than the issue going unnoticed. 

In this case, I was also thinking about it before I noticed @agisilaos comment. I came to the conclusion that I can't imagine that the cell doesn't exist when this method is called.